### ISSUE TITLE
[Todoist] Search Tasks render tasks for a short amount of time

### ISSUE THREAD

**Issue by tec1106** [raycast/extensions#6049]
### Extension

https://www.raycast.com/thomaslombart/todoist

### Description

8 out of 9 times I try to search for tasks, they are rendered for a few milliseconds and then they disappear, showing the 'No results' screen. If I try enough times, they will stay. If I then search again, the same thing happens all over again. 

### Current Behaviour

Search for tasks renders the tasks and then they quickly disappear.

### Expected Behaviour

For the tasks to stay rendered.

**Comment by thomaslombart** [comment]
That's weird, I can't reproduce. Could you link a recording of what you're getting?

**Comment by tec1106** [comment]
@thomaslombart Absolutely: https://share.cleanshot.com/kvlj8YqR

Some more context: I'm on an M1 Pro MacBook pro, a Clean install of Raycast, and I have hotkeys to activate search (the result is the same whether or not I use hotkeys)

**Comment by thomaslombart** [comment]
Thanks for this! Are you able to see tasks in other commands (today, upcoming, projects)?

**Comment by tec1106** [comment]
Yes. This only seems to happen under 'search task'

**Comment by tec1106** [comment]
I've also tried restarting the computer, reinstalling the extension, and even regenerating a new token - in case that might have been relevant somehow.

**Comment by tec1106** [comment]
Update 2: I've added a console count under the search function. I have 5 tasks. There is some erratic behavior that seem to be related. Sometimes it calls the function 6 times, sometimes 7 and sometimes 5. 

Generally, it's shows the tasks successfully when the function is called 5, or 7 times.

`15:20:41.392 Searching for tasks: (failed)
15:20:41.402 Search function: 1
15:20:41.425 Search function: 2
15:20:41.425 Search function: 3
15:20:41.615 Search function: 4
15:20:41.619 Search function: 5
15:20:41.926 Search function: 6
15:20:43.442 Searching for tasks: (failed)
15:20:43.452 Search function: 1
15:20:43.473 Search function: 2
15:20:43.473 Search function: 3
15:20:43.672 Search function: 4
15:20:43.674 Search function: 5
15:20:43.731 Search function: 6
15:20:45.025 Searching for tasks: (failed)
15:20:45.035 Search function: 1
15:20:45.056 Search function: 2
15:20:45.056 Search function: 3
15:20:45.245 Search function: 4
15:20:45.250 Search function: 5
15:20:45.501 Search function: 6
15:20:46.974 Searching for tasks: (failed)
15:20:46.985 Search function: 1
15:20:47.008 Search function: 2
15:20:47.008 Search function: 3
15:20:47.158 Search function: 4
15:20:47.162 Search function: 5
15:20:47.228 Search function: 6
15:20:48.516 Searching for tasks: (success)
15:20:48.526 Search function: 1
15:20:48.547 Search function: 2
15:20:48.547 Search function: 3
15:20:48.995 Search function: 4
15:20:49.001 Search function: 5`

**Comment by thomaslombart** [comment]
Thanks for debugging it. I'll take a look at that.

**Comment by thomaslombart** [comment]
Since you have the fork locally, what happens if you replace `filteredTasks` in the `List` component by `tasks?` (like below):

```tsx
  return (
    <List
      searchBarPlaceholder={placeholder}
      {...(projects ? { searchBarAccessory } : {})}
      isLoading={isLoadingTasks || isLoadingProjects}
    >
      {tasks?.map((task) => (
        <TaskListItem
          key={task.id}
          task={task}
          mode={projectId ? ViewMode.project : ViewMode.search}
          mutateTasks={mutateTasks}
          {...(projects ? { projects } : {})}
        />
      ))}
    </List>
  );
```

I'm wondering if something's happening with the filtering.

**Comment by tec1106** [comment]
Same behaviour

**Comment by tec1106** [comment]
Update: After completing a task in Raycast I no longer experience this behavior. I cannot replicate the behavior by completing the task. But it still flickers before stopping, which still indicates something is not quite right. See video: https://share.cleanshot.com/c3BJqKN8

**Comment by thomaslombart** [comment]
Hey, I'm working on the overhaul of the extension that I should ship in the next few days or weeks. This will probably fix your issue. Stay tuned! 📻

**Comment by thomaslombart** [comment]
Fixed: #6498
